### ISSUE TITLE
Matrix: Create a class to store primitive values in matrix (like a like std::vector)

### ISSUE THREAD

**Issue by kenorb** [EA31337/EA31337-classes#310]
- We need a class (e.g. `Matrix`) to store 3D shape values (equivalent to `double data[][][]`).
- If possible, it can support multiple dimensions (2D, 3D, 4D, 5D).
- Create a test to test the functionality.
- We should be able to traverse through all the elements (can be simple `for`, e.g. `for (int x = 0; x < data.GetRange(0); x++)`. 1 loop for 1 dimension, 2 loops for 2, and so on.

Similar-like implementation:

```
template<typename T>
class objvector : public CArrayObj {
public:
   T operator[](const int index) const { return At(index); }
};
```
but we don't want to use `CArrayObj`. It's going to be custom implementation.

Example usage:

```
// Define 3D matrix upfront.
Matrix<double> matrix(5, 5, 5); // Constructor calling SetShape()?
matrix.Fill(0); // Fill all the elements with 0. Could be also be called: matrix.Init() or matrix.Initialize().
double sum = matrix.Sum(); // Sums all the elements. It should be 0.
matrix.FillRandom(); // Fill all the elements with random (int) values.
matrix.FillRandom(0, 10); // Fill all the elements with random values between 0 and 10.
matrix[1][1][1] = 1.1;
matrix[2][2][2] = 1.2;
matrix[3][3][3] = 3;
double value = matrix[2][2][2]; // 1.2; Alternatively like: matrix.At(2, 2, 2) or something similar.
matrix.Add(2); // Add 2 to all values. Can we also do: matrix += 2 with operator?
matrix.Sub(2); // Substract 2 from all values. Can we also do: matrix -= 2?
matrix.Mul(2); // Multiple all values by 2.
matrix.Div(2); // Divide all values by 2.
// Lowest/Highest/Avg/Med
Print("Lowest: ", matrix.Min());
Print("Highest: ", matrix.Max());
Print("Average: ", matrix.Avg());
Print("Median: ", matrix.Med());
// ...
double size = matrix.GetSize(); // =125; Get number of all elements.
double dims = matrix.GetShape(); // = 3; Get number of dimensions (shape). One of: GetShape() or GetDimensions()
double dim1 = matrix.GetRange(0); // =5; Get number of elements in the 1st dimension.
double dim2 = matrix.GetRange(1); // =5; Get number of elements in the 2nd dimension.
double dim3 = matrix.GetRange(2); // =5; Get number of elements in the 3rd dimension.

// Define 4D matrix.
Matrix<float> *shape4d(2, 2, 2, 2); // Constructor calling SetShape()?

// Initialize empty matrix data of ints.
Matrix<int> *data;
data.SetShape(3, 3, 3); // Or maybe: data.SetShape(3, 3); which is 3x3?
data[1][1][1] = 10;
data.SetShape(3, 4, 5); // Can we support changing of shape? Can we support dynamic ranges (not all dimensions having the same size).
data[2][2][2] = 10;
data.SetShape(2, 2, 2); // Can we support downsizing of matrix?
data[1][1][1] = 10;

// Can we support initializing by data?
Matrix<int> *data2(2, 2) = {{0, 1, 2}, {1, 1, 1}};
```

Further calculation support by another matrix:

```
Matrix<double> matrix1(3, 3, 3);
Matrix<double> matrix2(3, 3, 3);
Matrix<double> matrix3(3, 3, 3);
matrix1.Fill(0);
matrix2.Fill(1);
matrix3.Fill(-1);
matrix1.Mul(matrix2); // Each element of matrix1 is multiplied by matrix2 elements (1 by 1). So: 0*1=0
matrix1.Add(matrix2); // Each element of matrix1 is added by corresponding matrix2 elements. Foreach: 0+1=1. Now all elements should be 1.
matrix1.Sub(matrix2); // Each element of matrix1 is subtracted/deducated by corresponding matrix2 elements. Foreach: 1-1=0. Now all elements should be 0 again.
matrix2.Div(matrix2); // Each element of matrix2 is divided by corresponding matrix2 elements. Foreach: 1/1=1. Now all elements should be 1.
matrix3.Abs(); // Make all elements absolute (convert negatives into positives). All elements should be 1 now.
```

Additional fill methods:
```
Matrix<int> matrix(3, 3);
matrix.FillPosAdd(); // Fill elements with corresponding x+y positions (x=x+y). [1][1]=2,[2][2]=4,[1][2]=3
matrix.FillPosMul(); // Fill elements with corresponding x*y positions (x=x*y). [1][1]=1,[2][2]=4,[1][2]=2
// For 3D, the same logic: x=x+y+z or x=x*y*z and so on.
```

Resources:

- https://stackoverflow.com/q/393556
- https://www.mql5.com/en/docs/standardlibrary/datastructures/carrayobj
- https://www.mql5.com/en/forum/34637
- https://en.cppreference.com/w/cpp/container/vector

**Comment by kenorb** [comment]
- `FillPosAdd()` and `FillPosMul()` implemented in a03d2a61b435429b5601905b8f65e13b51cabe49.

**Comment by kenorb** [comment]
Fixed in #339.
